filebeat/harvester/log.go

```diff
@@ -79,45 +107,46 @@ func (h *Harvester) Harvest() {
 	// TODO: NewLineReader uses additional buffering to deal with encoding and testing
 	//       for new lines in input stream. Simple 8-bit based encodings, or plain
 	//       don't require 'complicated' logic.
-	timedIn := newTimedReader(h.file)
-	reader, err := encoding.NewLineReader(timedIn, enc, h.Config.BufferSize)
+	config := h.Config
+	readerConfig := logFileReaderConfig{
+		forceClose:         config.ForceCloseFiles,
+		maxInactive:        h.ProspectorConfig.IgnoreOlderDuration,
+		backoffDuration:    config.BackoffDuration,
+		maxBackoffDuration: config.MaxBackoffDuration,
+		backoffFactor:      config.BackoffFactor,
+	}
+	reader, err := createLineReader(h.file, enc, config.BufferSize, readerConfig, config.Multiline)
 	if err != nil {
 		logp.Err("Stop Harvesting. Unexpected encoding line reader error: %s", err)
 		return
 	}
 
-	// XXX: lastReadTime handling last time a full line was read only?
-	//      timedReader provides timestamp some bytes have actually been read from file
-	lastReadTime := time.Now()
-
 	for {
 		// Partial lines return error and are only read on completion
-		text, bytesRead, err := readLine(reader, &timedIn.lastReadTime)
-
+		text, bytesRead, err := readLine(reader)
```

Comment:
with us removing partial line support, readLine did not return before anymore. Retrying + backoff is moved to first stage in reader (which more resembles mod-time logic in prospector) to not confuse multiline reader state machine.